File: packages/java/anip-core/src/main/java/dev/anip/core/AuditEntry.java
package dev.anip.core;

import java.util.List;
import java.util.Map;

/**
 * A single audit log record.
 */
public class AuditEntry {

    private int sequenceNumber;
    private String timestamp;
    private String capability;
    private String tokenId;
    private String issuer;
    private String subject;
    private String rootPrincipal;
    private Map<String, Object> parameters;
    private boolean success;
    private Map<String, Object> resultSummary;
    private String failureType;
    private CostActual costActual;
    private List<String> delegationChain;
    private String invocationId;
    private String clientReferenceId;
    private String previousHash;
    private String signature;
    private String eventClass;
    private String retentionTier;
    private String expiresAt;
    private boolean storageRedacted;
    private String entryType;
    private Map<String, Object> streamSummary;

    public AuditEntry() {}

    public int getSequenceNumber() { return sequenceNumber; }
    public void setSequenceNumber(int sequenceNumber) { this.sequenceNumber = sequenceNumber; }

    public String getTimestamp() { return timestamp; }
    public void setTimestamp(String timestamp) { this.timestamp = timestamp; }

    public String getCapability() { return capability; }
    public void setCapability(String capability) { this.capability = capability; }

    public String getTokenId() { return tokenId; }
    public void setTokenId(String tokenId) { this.tokenId = tokenId; }

    public String getIssuer() { return issuer; }
    public void setIssuer(String issuer) { this.issuer = issuer; }

    public String getSubject() { return subject; }
    public void setSubject(String subject) { this.subject = subject; }

    public String getRootPrincipal() { return rootPrincipal; }
    public void setRootPrincipal(String rootPrincipal) { this.rootPrincipal = rootPrincipal; }

    public Map<String, Object> getParameters() { return parameters; }
    public void setParameters(Map<String, Object> parameters) { this.parameters = parameters; }

    public boolean isSuccess() { return success; }
    public void setSuccess(boolean success) { this.success = success; }

    public Map<String, Object> getResultSummary() { return resultSummary; }
    public void setResultSummary(Map<String, Object> resultSummary) { this.resultSummary = resultSummary; }

    public String getFailureType() { return failureType; }
    public void setFailureType(String failureType) { this.failureType = failureType; }

    public CostActual getCostActual() { return costActual; }
    public void setCostActual(CostActual costActual) { this.costActual = costActual; }

    public List<String> getDelegationChain() { return delegationChain; }
    public void setDelegationChain(List<String> delegationChain) { this.delegationChain = delegationChain; }

    public String getInvocationId() { return invocationId; }
    public void setInvocationId(String invocationId) { this.invocationId = invocationId; }

    public String getClientReferenceId() { return clientReferenceId; }
    public void setClientReferenceId(String clientReferenceId) { this.clientReferenceId = clientReferenceId; }

    public String getPreviousHash() { return previousHash; }
    public void setPreviousHash(String previousHash) { this.previousHash = previousHash; }

    public String getSignature() { return signature; }
    public void setSignature(String signature) { this.signature = signature; }

    public String getEventClass() { return eventClass; }
    public void setEventClass(String eventClass) { this.eventClass = eventClass; }

    public String getRetentionTier() { return retentionTier; }
    public void setRetentionTier(String retentionTier) { this.retentionTier = retentionTier; }

    public String getExpiresAt() { return expiresAt; }
    public void setExpiresAt(String expiresAt) { this.expiresAt = expiresAt; }

    public boolean isStorageRedacted() { return storageRedacted; }
    public void setStorageRedacted(boolean storageRedacted) { this.storageRedacted = storageRedacted; }

    public String getEntryType() { return entryType; }
    public void setEntryType(String entryType) { this.entryType = entryType; }

    public Map<String, Object> getStreamSummary() { return streamSummary; }
    public void setStreamSummary(Map<String, Object> streamSummary) { this.streamSummary = streamSummary; }
}

File: packages/csharp/src/Anip.Core/AuditEntry.cs
using System.Text.Json.Serialization;

namespace Anip.Core;

public class AuditEntry
{
    [JsonPropertyName("sequence_number")]
    public int SequenceNumber { get; set; }

    [JsonPropertyName("timestamp")]
    public string Timestamp { get; set; } = "";

    [JsonPropertyName("capability")]
    public string Capability { get; set; } = "";

    [JsonPropertyName("token_id")]
    public string? TokenId { get; set; }

    [JsonPropertyName("issuer")]
    public string? Issuer { get; set; }

    [JsonPropertyName("subject")]
    public string? Subject { get; set; }

    [JsonPropertyName("root_principal")]
    public string? RootPrincipal { get; set; }

    [JsonPropertyName("parameters")]
    public Dictionary<string, object>? Parameters { get; set; }

    [JsonPropertyName("success")]
    public bool Success { get; set; }

    [JsonPropertyName("result_summary")]
    public Dictionary<string, object>? ResultSummary { get; set; }

    [JsonPropertyName("failure_type")]
    public string? FailureType { get; set; }

    [JsonPropertyName("cost_actual")]
    public CostActual? CostActual { get; set; }

    [JsonPropertyName("delegation_chain")]
    public List<string>? DelegationChain { get; set; }

    [JsonPropertyName("invocation_id")]
    public string? InvocationId { get; set; }

    [JsonPropertyName("client_reference_id")]
    public string? ClientReferenceId { get; set; }

    [JsonPropertyName("task_id")]
    public string? TaskId { get; set; }

    [JsonPropertyName("parent_invocation_id")]
    public string? ParentInvocationId { get; set; }

    [JsonPropertyName("upstream_service")]
    public string? UpstreamService { get; set; }

    [JsonPropertyName("previous_hash")]
    public string? PreviousHash { get; set; }

    [JsonPropertyName("signature")]
    public string? Signature { get; set; }

    [JsonPropertyName("event_class")]
    public string? EventClass { get; set; }

    [JsonPropertyName("retention_tier")]
    public string? RetentionTier { get; set; }

    [JsonPropertyName("expires_at")]
    public string? ExpiresAt { get; set; }

    [JsonPropertyName("storage_redacted")]
    public bool StorageRedacted { get; set; }

    [JsonPropertyName("entry_type")]
    public string? EntryType { get; set; }

    [JsonPropertyName("grouping_key")]
    public Dictionary<string, string>? GroupingKey { get; set; }

    [JsonPropertyName("aggregation_window")]
    public Dictionary<string, string>? AggregationWindow { get; set; }

    [JsonPropertyName("aggregation_count")]
    public int AggregationCount { get; set; }

    [JsonPropertyName("first_seen")]
    public string? FirstSeen { get; set; }

    [JsonPropertyName("last_seen")]
    public string? LastSeen { get; set; }

    [JsonPropertyName("representative_detail")]
    public string? RepresentativeDetail { get; set; }

    [JsonPropertyName("stream_summary")]
    public Dictionary<string, object>? StreamSummary { get; set; }
}
